### DISCUSSION TITLE
How to remove "full" alignment option?

### DISCUSSION THREAD

**Discussion by scottfennell-toptal** [WordPress/gutenberg#47359] [Q&A]
See screenshot.  I don't want to support a "full" width in my theme.  What is the themes.json entry, or php filter, or wp js api approach for filtering those buttons out?

<img width="2048" alt="Screen Shot 2023-01-23 at 2 49 28 PM" src="https://user-images.githubusercontent.com/34286895/214135673-aab50587-2666-4c1a-96e5-f387e125782b.png">

**Comment by devfle** [comment]
Hey, you can use the following filter hook for this: https://developer.wordpress.org/block-editor/reference-guides/filters/block-filters/#blocks-registerblocktype

Example:

```javascript
function removeAlignOption( settings, name ) {
    if (!settings) return settings;

    if (!settings?.supports) settings.supports = {};

    if (settings.supports?.align) settings.supports.align = false;

    return settings;
}

wp.hooks.addFilter(
    'blocks.registerBlockType',
    'my-plugin/remove-align/all-blocks',
    removeAlignOption
);
```

**Comment by scottfennell-toptal** [comment]
Thank you for the response!  I'm having trouble getting this to execute.  I would assume all this sort of stuff should be wrapped in wp.domReady, but maybe I'm wrong about that?

Here is my attempt to follow your suggestion:

```
wp.domReady( function() {

	// This guy logs.
	console.log( 29 );

	function removeAlignOption( settings, name ) {
	
		// This guy does not log.
		console.log( 31 );

	}

	wp.hooks.addFilter(
		'blocks.registerBlockType',
		'my-plugin/remove-align/all-blocks',
		removeAlignOption
	);

});
```

Aside -- how did you get JS syntax highlighting in your comment?

**Reply by devfle** [reply]
The filter must always return the settings obj like in my example. Otherwise there will be an issue. You do not need to wrap it inside wp.domReady(); Just enqueue your script at: https://developer.wordpress.org/reference/hooks/wp_enqueue_scripts/

For Syntax highlighting just write next to the ```javascript

**Comment by scottfennell-toptal** [comment]
Thanks.

Sure, I understand a filter must always return the value passed into it.

I've attempted this again, with slightly better results outside of the domReady.  But I still am not getting the expected behavior, see:

```javascript
	function removeAlignOption( settings, name ) {
		
		if( name != 'core/image' ) { return settings; }
	
		if( ! settings ) return settings;

		console.log( 'this guy logs' );

		if ( ! settings?.supports ) { 
			settings.supports = {};
		}

		if ( settings.supports?.align ) { 
			console.log( 'this guy doesnt log' );
			settings.supports.align = false;
		}

		return settings;

	}

	wp.hooks.addFilter(
		'blocks.registerBlockType',
		'my-plugin/remove-align/all-blocks',
		removeAlignOption
	);
```

So it seems that for some reason the settings.supports enters this hook with no support for align at all, and I'm not sure why.

**Reply by devfle** [reply]
Hello again,

I just checked again in more detail and indeed it is the case that the positioning on the image block cannot be manipulated via the Supports API. With the Group Block, for example, this is possible. I first thought that your question also refers to other blocks. I have also not found a way to remove this one function via an official API.

But I think it would be good if this could be manipulated via e.g. the Supports API. Maybe if there is no official way to do so, we could create an issue?
